tds-itemselection-impl/src/test/java/msb/MsbAssessmentSelectionServiceTest.java

```diff
@@ -0,0 +1,268 @@
+        }
+        ItemGroup itemGroup1 = itemGroupBuilder
+                .withItems(testItems1)
+                .build();
+        List<ItemGroup> itemGroups1 = new ArrayList<>();
+        itemGroups1.add(itemGroup1);
+        ItemPool itemPool1 = itemPoolBuilder
+                .withItemGroups(itemGroups1)
+                .withItems(testItems1)
+                .withSiblingItems(testItems1)
+                .build();
+
+        TestSegment testSegment1 = testSegmentBuilder.build();
+
+        return testSegments;
+    }
+
+    //region filterItemCandidates tests
+
+    @Test
+    public void filterItemCandidatesWithFixedFormFilterShouldReturnTwoResults() {
+        // Arrange
+        List<ItemCandidatesData> itemCandidatesData = retrieveItemCandidatesData();
+
+        // Act
+        List<ItemCandidatesData> filterredItemCandidates = msbAssessmentSelectionService.filterItemCandidatesByAlgorithm(itemCandidatesData, "fixedform");
+
+        // Assert
+        Assert.assertTrue(filterredItemCandidates.size() == 2);
+        Assert.assertTrue(filterredItemCandidates.get(0).getAlgorithm() == "fixedform");
+        Assert.assertTrue(filterredItemCandidates.get(1).getAlgorithm() == "fixedform");
+        Assert.assertTrue(filterredItemCandidates.get(0).getSegmentKey() == "(SBAC_PT)SBAC-MSB-IRP-Perf-MATH-7-Summer-2015-2016");
+        Assert.assertTrue(filterredItemCandidates.get(1).getSegmentKey() == "(SBAC_PT)SBAC-MSB-IRP-Perf-MATH-11-Summer-2015-2016");
+    }
+
+    @Test
+    public void filterItemCandidatesWithFixedFilterShouldReturnNoResults() {
+        // Arrange
+        List<ItemCandidatesData> itemCandidatesData = retrieveItemCandidatesData();
+
+        // Act
+        List<ItemCandidatesData> filterredItemCandidates = msbAssessmentSelectionService.filterItemCandidatesByAlgorithm(itemCandidatesData, "fixed");
+
+        // Assert
+        Assert.assertTrue(filterredItemCandidates.isEmpty());
+    }
+
+    @Test
+    public void filterItemCandidatesWithFixedFormUpperCaseFilterShouldReturnTwoResults() {
+        // Arrange
+        List<ItemCandidatesData> itemCandidatesData = retrieveItemCandidatesData();
+
+        // Act
+        List<ItemCandidatesData> filterredItemCandidates = msbAssessmentSelectionService.filterItemCandidatesByAlgorithm(itemCandidatesData, "FiXedFoRM");
+
+        // Assert
+        Assert.assertTrue(filterredItemCandidates.size() == 2);
+        Assert.assertTrue(filterredItemCandidates.get(0).getAlgorithm() == "fixedform");
+        Assert.assertTrue(filterredItemCandidates.get(1).getAlgorithm() == "fixedform");
+        Assert.assertTrue(filterredItemCandidates.get(0).getSegmentKey() == "(SBAC_PT)SBAC-MSB-IRP-Perf-MATH-7-Summer-2015-2016");
+        Assert.assertTrue(filterredItemCandidates.get(1).getSegmentKey() == "(SBAC_PT)SBAC-MSB-IRP-Perf-MATH-11-Summer-2015-2016");
+    }
+
+    @Test
+    public void filterItemCandidatesWithExcludedAlgorithmTypeFilterShouldReturnNoResults() {
```

Comment:
I cannot see the difference between this test and `filterItemCandidatesWithNoCandidatesShouldReturnNoResults()`.  Can we either make it more clear in the code or add comments?  Assuming that they both test something different.